### ISSUE TITLE
[Feature Request]: Handling winget upgrade reboots aka exit code 3010

### ISSUE THREAD

**Issue by RNalivaika** [Romanitho/Winget-AutoUpdate#765]
### The request

Some software installs and upgrades (for example "Microsoft.VCRedist.2015+.x86") exit with code 3010. That means reboot required to complete installation.
I would like some options in handling that. For example, define x number of days before reboot is forced, pop up to inform the user about pending reboot and give opportunity to reboot now or snooze for x number of minutes.

### Is your feature request related to a problem?

```txt
Some software installs and upgrades (for example "Microsoft.VCRedist.2015+.x86") exit with code 3010. That means reboot required to complete installation.
I would like some options in handling that. For example, define x number of days before reboot is forced, pop up to inform the user about pending reboot and give opportunity to reboot now or snooze for x number of minutes.
```

**Comment by AndrewDemski-ad-gmail-com** [comment]
Hey,

Ideas:
Toast Notification about 3010 - could be done
triggering actual reboot from within WAU - definitely not

IMHO this would be going too far.
Keep in mind that WAU is used in SOHO and enterprise scenarios.
It is up to the user (or administrator) to decide when such action should occur (preferably within a maintenance window if such a thing is defined).

BR
AD

**Comment by FaserF** [comment]
I do not think that this is possible, since we cannot fetch the Exit Code with Powershell. Winget only displays it as text with the information from above. 
But I think Andrew is right. Using it in a company makes handling return codes "easy", in a private scenario I do not think we need such feature.

**Comment by KnifMelti** [comment]
Most of that can be checked via https://github.com/Romanitho/Winget-AutoUpdate/blob/e31afa874900d581003fa3c15bbda71dc20d3f6b/Sources/Winget-AutoUpdate/functions/Test-PendingReboot.ps1

**Comment by AndrewDemski-ad-gmail-com** [comment]
OMG I completely forgot about that script.. shame on me..

**Comment by RNalivaika** [comment]
So maybe after Winget-AU is finished with scheduled update install, the pending reboot script could run immediately.
If pending reboot is detected, a pop up message could pop up giving the user options to either "dismiss" or "reboot-now".

ps: suggestion for expanding pending reboot script would be to include another registry path:
-Path 'HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager' -Name 'PendingFileRenameOperations'
